mining/src/manager.rs

```diff
@@ -717,10 +717,10 @@ impl MiningManager {
             let _swo = Stopwatch::<60>::with_threshold("revalidate update_revalidated_transaction op");
             for (transaction, validation_result) in chunk {
                 let transaction_id = transaction.id();
-                // Only consider transactions still being in the mempool since during the validation some might have been removed.
-                if mempool.update_revalidated_transaction(transaction) {
-                    match validation_result {
-                        Ok(()) => {
+                match validation_result {
+                    Ok(()) => {
+                        // Only consider transactions still being in the mempool since during the validation some might have been removed.
+                        if mempool.update_revalidated_transaction(transaction) {
```

Comment:
style suggestion
```rust
                    // Only consider transactions still being in the mempool since during the validation some might have been removed.
                    Ok(()) if mempool.update_revalidated_transaction(transaction) => {
                        // A following transaction should not remove this one from the pool since we process in a topological order.
                        // Still, considering the (very unlikely) scenario of two high priority txs sandwiching a low one, where
                        // in this case topological order is not guaranteed since we only considered chained dependencies of
                        // high-priority transactions, we might wrongfully return as valid the id of a removed transaction.
                        // However, as only consequence, said transaction would then be advertised to registered peers and not be
                        // provided upon request.
                        valid_ids.push(transaction_id);
                        valid += 1;
                    }
                    Ok(()) => {
                        other += 1;
                    }
```

Reply:
This can't be done because it consumes `transaction` which is used in the err branches